js/interface.js

```diff
@@ -285,6 +285,47 @@ function initLinkProvider(item) {
   linkPromises.push(linkActionProvider);
 }
 
+function initListener(provider, item) { 
+  window.addEventListener('message', function onMessage(event) {
+    if (event.data === 'cancel-button-pressed') {
+      switch (provider) {
+        case 'icon':
+          onIconClose(item);         
+          break;
+        case 'image':
+          onImageClose(item);
+          break;
+      }
+
+      window.removeEventListener('message', onMessage);
```

Comment:
`onMessage` is not defined. You should save it as a function so you can properly remove the listener, e.g.:

```js
var onMessage = function () {
  window.removeEventListener('message', onMessage);
};

window.addEventListener('message', onMessage);
```


```

Reply:
@squallstar It is defined at this [line](https://github.com/Fliplet/fliplet-widget-list-small-thumbs/pull/41/files#diff-9fa998c3e15bcce1e37aec501c4937bdR289). And it is possible to work with it without declaring another variable.
And for this function to work correctly we need to pass threw it an `item` object  and `event` as well so we could transform UI correctly, but for that, we would need to do something like that: 
```
var onMessage = function (event, item) {
 // Doing what we need
}

window.addEventListener('message', function (event) { onMessage(event, item) })
```
But in this case we couldnt remove event listener because it is nameless function is our listener but not a `onMessage` function.

Reply:
@YaroslavOvdii no, that's incorrect. You are only passing a named function as an attribute to a function, it's not being defined so you won't be able to remove that afterwards.

See examples that I just ran in the console:

![image](https://user-images.githubusercontent.com/574210/78237064-1fe54380-74db-11ea-9694-8609eeadce70.png)

This is pretty basic JS as you can see.

Reply:
@squallstar Yes, you are right, but that is why we are using `window.removeEventListener('message', onMessage)` inside the scope of the `onMessage` function where it exists.
![image](https://user-images.githubusercontent.com/53430352/78243875-3ac0b380-74ed-11ea-87fa-57ecef884a31.png)
And that is how we are removing listeners from firing again.

Reply:
Ok I see, I guess that works then.